services/web/server/src/simcore_service_webserver/studies_dispatcher/_models.py

```diff
@@ -58,3 +49,23 @@ def create_from_db(cls, row: RowProxy) -> "ViewerInfo":
             input_port_key=row["service_input_port"],
             is_guest_allowed=row["is_guest_allowed"],
         )
+
+
+class ServiceParams(BaseModel):
+    viewer_key: ServiceKey
+    viewer_version: ServiceVersion
+
+    @property
+    def footprint(self) -> str:
```

Comment:
From the usage in the code this looks like an alias for `__str__`.

Reply:
not really. It is more like a hash. It is to determine a "footprint" of the class info. Then it is used to generate uuid for the project i.e.
```python
project_uid: ProjectID = compose_uuid_from(user.id, viewer.footprint, download_link)
```
this way if the same user asks the same viewer with the same file, it will get the already existing project

Reply:
footprint reminds me about resource consumptions, so I think it's misleading. But OK.

Reply:
@GitHK provide some alternatives ...

What about checksum?